src/features/wordDisplay/buttonWord.tsx

```diff
@@ -492,11 +496,11 @@ export const ButtonToken = ({
         ...(isInLinkWithCurrentlyHoveredToken && !isSelectedInEditedLink ? hoverSx : {}),
         ...(fillWidth ? { width: '100%' } : {})
       })}
-      onMouseEnter={!!hoverHighlightingDisabled || (!!editedLink && !isSelectedInEditedLink) ? () => {} : () => dispatch(hover(token))}
+      onMouseEnter={!!hoverHighlightingDisabled || (!editedLink && isSelectedInEditedLink) ? () => {} : () => dispatch(hover(token))}
```

Comment:
@michael-kitchin This is the fix for the bug in this story. Everything else is just cleanup and tidying the code.